### PR TITLE
Document the test-only intentions of this package

### PR DESCRIPTION
**Pull request by dsnet**
This package was never intended to be used in production code.
Document the expectation that this package was only intended
to be used for writing tests.

### CODE REVIEW

File: cmp/compare.go [hunk 1]
```diff
@@ -6,6 +6,9 @@
 //
 // This package is intended to be a more powerful and safer alternative to
 // reflect.DeepEqual for comparing whether two values are semantically equal.
+// It is intended to only be used in tests, as cmp may panic if it cannot
+// compare the values. It's propensity towards panicking means that it is
```
Comment (neild) [inline cmp/compare.go]: Its

File: cmp/compare.go [hunk 2]
```diff
@@ -6,6 +6,9 @@
...
+// unsuitable for production environments where a spurious panic may be fatal.
```
Comment (neild) [inline cmp/compare.go]: Maybe also mention that performance is not really a goal?

File: cmp/compare.go (hunk 1 above)
Comment (dsnet) [reply cmp/compare.go]: Kept it in untracted form for the first occurrence. Used contracted form for later occurences.

File: cmp/compare.go (hunk 2 above)
Comment (dsnet) [reply cmp/compare.go]: Done.

File: cmp/compare.go [hunk 3]
```diff
@@ -6,6 +6,10 @@
 //
 // This package is intended to be a more powerful and safer alternative to
 // reflect.DeepEqual for comparing whether two values are semantically equal.
+// It is intended to only be used in tests, as performance is not a goal and
+// it may panic if it cannot compare the values. It's propensity towards
```
Comment (neild) [inline cmp/compare.go]: s/It's/Its/

File: cmp/compare.go (hunk 1 above)
Comment (dsnet) [reply cmp/compare.go]: Oh, you were probably referring to the "it's", which is incorrect grammar. Sigh, the inability to highlight a specific word in GitHub PR is confusing.

File: cmp/compare.go (hunk 3 above)
Comment (dsnet) [reply cmp/compare.go]: Just realized this. Already fixed.

**Pull request merged**